app/controllers/api/subscriptions_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class Api::SubscriptionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    if params[:stripeToken] && !params[:stripeToken].empty?
+      customer_id = get_customer(params[:stripeToken])
+      subscription = Stripe::Subscription.create({ customer: customer_id, plan: "dns_subscription" })
+
+      Rails.env.test? && test_env(customer_id, subscription)
+      payment_status(subscription)
+    else
+      render json: { message: "Transaction was NOT successful. There was no token provided..." }, status: 422
+    end
+  end
+
+  def get_customer(stripeToken)
+    customer = Stripe::Customer.list(email: current_user.email).data.first
+    customer ||= Stripe::Customer.create({ email: current_user.email, source: stripeToken })
+    customer.id
+  end
+
+  def test_env(customer_id, subscription)
+    invoice = Stripe::Invoice.create({ customer: customer_id, subscription: subscription.id, paid: true })
+    subscription.latest_invoice = invoice.id
+  end
+
+  def payment_status(subscription)
+    status = Stripe::Invoice.retrieve(subscription.latest_invoice).paid
+
+    if status
+      current_user.update_attribute(:subscriber, true)
```

Comment:
The user will become a lifetime subscriber?

Reply:
What happens if the user is already a subscriber? I think it should not be possible for already lifetime subscribers to but another lifetime subscription.

Reply:
:grimacing:

Reply:
Regarding life-time subscription - We on front-end realised that if we're going to have a ongoing subscription, then we'd have to implement a way to cancel it as well. So we thought it would be easiest to subscribe and pay for 1 month at a time. Sorry that we forgot to bring that up...!